src/clr-angular/data/datagrid/datagrid.html

```diff
@@ -1,60 +1,71 @@
 <!--
-  ~ Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+  ~ Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
   ~ This software is released under MIT license.
   ~ The full license information can be found in LICENSE in the root directory of this project.
   -->
 
 <ng-content select="clr-dg-action-bar"></ng-content>
-<div class="datagrid" #datagrid>
-    <div class="datagrid-table-wrapper">
-      <div role="grid" class="datagrid-table">
-        <div role="rowgroup" class="datagrid-header">
-          <div role="row" class="datagrid-row">
-            <div class="datagrid-row-master datagrid-row-flex">
-              <div class="datagrid-row-sticky">
-                <!-- Sticky elements here -->
-              </div>
-              <div class="datagrid-row-scrollable">
-                <!--header for datagrid where you can select multiple rows -->
-                <div role="columnheader" class="datagrid-column datagrid-select datagrid-fixed-column"
-                     *ngIf="selection.selectionType === SELECTION_TYPE.Multi">
-                            <span class="datagrid-column-title">
-                                <input clrCheckbox type="checkbox" [(ngModel)]="allSelected"
-                                       [attr.aria-label]="commonStrings.selectAll">
-                            </span>
-                  <div class="datagrid-column-separator"></div>
-                </div>
-                <!-- header for datagrid where you can select one row only -->
-                <div role="columnheader" class="datagrid-column datagrid-select datagrid-fixed-column"
-                     *ngIf="selection.selectionType === SELECTION_TYPE.Single">
-                  <div class="datagrid-column-separator"></div>
-                </div>
-                <!-- header for single row action; only displayType if we have at least one actionable row in datagrid -->
-                <div role="columnheader" class="datagrid-column datagrid-row-actions datagrid-fixed-column"
-                     *ngIf="rowActionService.hasActionableRow">
-                  <div class="datagrid-column-separator"></div>
+<div class="datagrid-outer-wrapper">
+  <div class="datagrid-wrapper" [class.datagrid-detail-open]="detailService.isOpen">
```

Comment:
As I mentioned before, this breaks scrolling. Will sync up for ideas on how to handle this but am going to block on it for now.